index.js

```diff
@@ -337,7 +337,7 @@ function compareString(buffer, start, target) {
   if (start === -1) return null
   target = Buffer.isBuffer(target) ? target : Buffer.from(target)
   var tag = varint.decode(buffer, start)
-  if (tag & (TAG_MASK !== STRING)) return null
+  if ((tag & TAG_MASK) !== STRING) return null
```

Comment:
This definitely seemed like a mistake, and one of the new tests caught this.

Reply:
Yes this definitely looks incorrect!